src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -396,11 +374,9 @@ private async Task<EntityStateResponse<T>> ReadEntityStateAsync<T>(TaskHubClient
             {
                 string serializedState;
 
-                if (this.client.ServiceClient is AzureStorageOrchestrationService service
-                    && state.Input.StartsWith("http"))
+                if (state.Input.StartsWith("http"))
```

Comment:
The blob compression stuff is specific to the Azure Storage provider, so it seems odd to have logic for handling it at this layer. We may want to consider refactoring how we handle this in a future PR.